### ISSUE TITLE
[BUG] Current develop dependencies broken

### ISSUE THREAD

**Issue by finwe** [barbushin/php-imap#599]
**Environment (please complete the following information):**
 - PHP IMAP version: dev-develop
 - PHP Version: 8.0.5
 - Type of execution: CLI with composer

**Describe the bug**

Current definition of PHP requirements are broken with current set of dependencies, Composer won't install.

For dependencies to support PHP 8, minimal PHP requirement has to be risen up to ^7.4, as tested in a fork https://github.com/finwe/php-imap/commit/dce42bc7be7339e51b4c549bdb67f10fe59f1d69

**To Reproduce**

- Checkout develop branch
- Run `composer update` with PHP 8

```
  Problem 1
    - Root composer.json requires sebastian/phpcpd ^4.1 -> satisfiable by sebastian/phpcpd[4.1.0].
    - sebastian/phpcpd 4.1.0 requires php ^7.1 -> your php version (8.0.5) does not satisfy that requirement.
  Problem 2
    - maglnet/composer-require-checker 2.0.0 requires php ^7.1 -> your php version (8.0.5) does not satisfy that requirement.
    - maglnet/composer-require-checker 2.1.0 requires php ^7.2 -> your php version (8.0.5) does not satisfy that requirement.
    - Root composer.json requires maglnet/composer-require-checker ^2.0 -> satisfiable by maglnet/composer-require-checker[2.0.0, 2.1.0].
```

**Comment by bapcltd-marv** [comment]
fixing the developer dependencies now, dropping php is not necessary.

**Comment by finwe** [comment]
So you are installing locked dependencies in versions not officially supported in higher PHP? Not what I would do, but I guess it's OK for dev dependencies.

**Comment by bapcltd-marv** [comment]
> So you are installing locked dependencies in versions not officially supported in higher PHP? Not what I would do, but I guess it's OK for dev dependencies.

All I did was run `composer require` for each of the problematic dependencies & made note to append the version constraints, not sure where you get the "not officially supported" thing from.

**Comment by finwe** [comment]
Oh, I see the commit now, sorry.

**Comment by Sebi94nbg** [comment]
This should be fixed now.

This change is included in the release [4.3.0](https://github.com/barbushin/php-imap/releases/tag/4.3.0) and newer.
